pkg/utils/host/host.go

```diff
@@ -80,16 +97,124 @@ func init() {
 	IsHostNetNs = isHostNamespace("net")
 }
 
-func workarounds() error {
-	// No memory limit for eBPF maps
-	if err := rlimit.RemoveMemlock(); err != nil {
+func checkKubectlDebug() error {
+	if HostRoot != "/host" || os.Getenv("KUBERNETES_PORT") == "" {
+		return nil
+	}
+
+	c, err := capability.NewPid2(0)
+	if err != nil {
+		return err
+	}
+	err = c.Load()
+	if err != nil {
+		return err
+	}
+	if c.Get(capability.EFFECTIVE, capability.CAP_SYS_ADMIN) {
+		// if we already have CAP_SYS_ADMIN, we don't need a workaround
+		return nil
+	}
+
+	_, err = os.Stat("/host/run/systemd/private")
+	if err != nil {
+		// if we don't have systemd, we cannot use the workaround
+		return nil
+	}
+	err = os.MkdirAll("/run/systemd", 0o755)
+	if err != nil {
 		return err
 	}
+	_, err = os.Stat("/run/systemd/private")
+	if err != nil {
+		err := os.Symlink("/host/run/systemd/private", "/run/systemd/private")
+		if err != nil {
+			return fmt.Errorf("linking /run/systemd/private: %w", err)
+		}
+	}
```

Comment:
I do that because the systemdDbus Golang package hard codes "/run/systemd/private" so I can't tell it to use "/host/run/systemd/private" instead. Normally the gadget DaemonSet uses the host volume `/run:/run` so that would not be necessary. But here, the gadget pod was started by "kubectl debug node", so the `/run` is not mounted from the host and is just a tmpfs in the container. So I make the symlink to make the systemdDbus Golang package work fine.

Yes, I can check the specific error code from Stat with `errors.Is(err, os.ErrNotExist)`.